### PR TITLE
[config] Miscellaneous cleaning of configuration variables.

### PR DESCRIPTION
**Pull request by ejgallego**
We perform a cleanup in `Coq_config` / `Envars`, consolidating handling of coqlib and removing unused / deprecated variables.

cc: #7149

### CODE REVIEW

File: ide/preferences.ml [hunk 1]
```diff
@@ -359,14 +366,22 @@ let text_font =
   in
   new preference ~name:["text_font"] ~init ~repr:Repr.(string)
 
+let is_standard_doc_url url =
+  let wwwcompatprefix = "http://www.lix.polytechnique.fr/coq/" in
```
Comment (Zimmi48) [inline ide/preferences.ml]: Since you are moving this function around, I think it is time to remove this compat thing and simplify the code.

Comment (ejgallego) [reply ide/preferences.ml]: Sure, I planned to do that however I got scared by the logic there.

**Comment by ejgallego** [comment]
@ppedrot you are the secondary for `lib` so maybe you could take this? `config/` has no owner, but it should be I guess the same than `configure` which is me, however there is no secondary.

**Comment by ppedrot** [comment]
I am not sure I am really proficient, but let's take ownership whatsoever because otherwise nobody will.

File: kernel/nativelib.ml [hunk 2]
```diff
@@ -37,7 +36,7 @@ let ( / ) = Filename.concat
 (* We have to delay evaluation of include_dirs because coqlib cannot be guessed
 until flags have been properly initialized *)
 let include_dirs () =
-  [Filename.get_temp_dir_name (); coqlib () / "kernel"; coqlib () / "library"]
+  [Filename.get_temp_dir_name (); Envars.coqlib () / "kernel"; Envars.coqlib () / "library"]
```
Comment (maximedenes) [inline kernel/nativelib.ml]: Why these changes?

Comment (ejgallego) [reply kernel/nativelib.ml]: I remove all `open Envars` so I can grep for who's using them, as all users need work.

For example `nativelib` will have to use a less adhoc compilation model to support Dune properly.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
Is this ready? If so I can merge quickly.

**Comment by ejgallego** [comment]
> Is this ready? If so I can merge quickly.

@ppedrot seems ready to me, I don't think there is a lot to see here tho, most of the PR is trivial, and would we discover some problem straightforward to revert.

**Comment by ejgallego** [comment]
Merci !
